docs/src/main/asciidoc/spring-cloud-square.adoc

```diff
@@ -0,0 +1,163 @@
+
+[source,java]
+----
+@RetrofitClient("serviceId")
+interface HelloClient {
+	@GET("/")
+	Call<String> hello();
+}
+----
+
+[source,java]
+----
+@RetrofitClient("serviceId")
+interface HelloClient {
+	@GET("/")
+	Mono<String> hello();
+}
+----
+
+`@RetrofitClient` takes a `value` (alias `name`) argument, that also indicates the `serviceId` which should be used during load-balancing.
+
+If you do not wish for the calls to be load-balanced, you should also use the `url` parameter, that allows you to set the full url, with host and port specified explicitly:
+
+[source,java]
+----
+@RetrofitClient(name = "localapp", url = "http://localhost:8080")
+	protected interface TestClient {
+
+		@GET("/hello")
+		Call<Hello> getHello();
+}
+----
+
+Apart from this, `@RetrofitClient` annotation allows you to specify a `qualifier` and pass a custom `configuration` class that you can use to provid or override OkHttp `Interceptor` beans, as in the following example:
+
+[source,java]
+----
+@RetrofitClient(name = "localapp", configuration = CustomRetrofitConfig.class)
+	protected interface TestClient {
+
+		@GET("/hello")
+		Call<Hello> getHello();
+}
+
+class CustomRetrofitConfig {
+
+		@Bean
+		public Interceptor interceptor1() {
+			return chain -> {
+				Request request = chain.request().newBuilder().addHeader(MYHEADER1, "myheader1value").build();
+				return chain.proceed(request);
+			};
+		}
+}
+----
+
+TIP: The configurations passed via `@RetrofitClient` and `@EnableRetrofitClients` annotations, behave similarly to the ones passed via `@LoadBalancerClient` and `@LoadBalancerClients` annotations. They are loaded into separate child contexts and should not be annotated with `@Configuration`.
+
+=== Load-balanced Retrofit clients
+
+In order to leverage Spring Cloud LoadBalancer for instance selection and instance data retrieval, make sure you instantiate either a `@LoadBalanced OkHttpClient.Builder` (for a blocking implementation) or a `@LoadBalanced Webclient.Builder` (for a reactive implementation) bean in a `@Configuration`-annotated class in your application:
+
+[source,java]
+----
+@Configuration
+@EnableRetrofitClients
+class OkHttpClientConfig {
+
+@Bean
+@LoadBalanced
+public OkHttpClient.Builder okHttpClientBuilder() {
+    return new OkHttpClient.Builder();
+    }
+}
+----
+
+[source,java]
+----
+@Configuration
+@EnableRetrofitClients
+class OkHttpClientConfig {
+
+@Bean
+@LoadBalanced
+public WebClient.Builder okHttpClientBuilder() {
+    return WebClient.builder();
+    }
+}
+----
+
+They will be used under the hood to run load-balanced HTTP requests.
+
+
+=== Retrofit Reactor support
+
+When `ReactorCallAdapterFactory` is on the classpath (provided by `retrofit2-reactor-adapter` dependency), we also instantiate a bean of this type, using available `Scheduler` if present. You can disable this functionality via properties by setting the value of `retrofit.reactor.enabled` to `false`.
```

Comment:
Change to:

When `ReactorCallAdapterFactory` is on the classpath (provided by `retrofit2-reactor-adapter` dependency), we also instantiate a bean of this type, by using an available `Scheduler` (if present). You can disable this functionality in properties by setting the value of `retrofit.reactor.enabled` to `false`.